src/Components/Filters.js

```diff
@@ -11,25 +11,52 @@ const Button = props => {
 };
 
 const Filters = props => {
-  const [level, setLevel] = useState();
+  const [selectedLevel, setLevel] = useState();
+  const [selectedGoal, setGoal] = useState();
 
   const onLevelChange = pressedLevel => {
-    const newLevel = pressedLevel === level ? undefined : pressedLevel;
+    const newLevel = pressedLevel === selectedLevel ? undefined : pressedLevel;
     setLevel(newLevel);
   };
+  const onGoalChange = pressedGoal => {
```

Comment:
I just duplicated code to create the second filter
I feel like we should refactor this when implementing the third one (rule of three)

Reply:
Ok I now have 3 filters working, with some nice duplicated code to refactor

Reply:
Have you ever worked with ramda (https://ramdajs.com/)?

It takes a bit to wrap your head around, but it's insanely useful for writing reusable functional (and immutable!) code (like filters).

As an example the filter code below could be rewritten to something like this:
```
const newData = R.pipe(
     R.filter(R.propEq('level', selectedLevel)),
     R.filter(R.propSatisfies('goals', R.includes(selectedGoal))),
     R.filter(R.propSatisfies('minimalPlayersNumber', R.lte(R.__, numberOfPlayers)))
)(initialData);
```

Reply:
I did'nt know it. It seems nice and could be used in `useEffect` below. My concern here is I'm not very happy with defining `  const [x, setX] = useState();` and `onXChange` for each filter and thinking about refactoring it.

But any implementation I imagine is a working abstraction bt makes it more difficult to understand (and modify or extend) than the actual duplication. I think I'll stick with the duplication :)

Reply:
Worst case there's always class components with a single `setState` function. You could probably leave the filtering in the render function as it only renders on state/prop updates anyway